packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+
+To approve an objective is to transfer ownership of those channels to the objective for the lifetime of the objective or until another more important objective is approved that takes ownership. This means that updates cannot be made to a channel in the scope of an approved objective that is still alive. This includes updates by any other objectives as well as explicit updates from the app.
+
+Approval of an objective requires all channels in its scope to be known to the wallet (i.e.) in it's store.
+
+### Rejection
+
+TODO
+
+### Cranking
+
+Aproved Objectives are _cranked_. This means that when certain events happen, the wallet attempts to make progress toward the Objective goal. It does this by reading relevant, scoped data from the database, deciding on an action to take (signing a state or a transaction) and then committing the result.
+
+(Here, committing means: reinserting into or patching the database; broadcasting states; broadcasting transactions; responding to the API request; emitting an event.)
+
+In practice, a _cranker_ is a function that accepts a store, a _response_ object, and a newly started database transaction context. It uses the transaction context to
+
+- lock one or more database rows;
+- read from the database;
+- writes to the database if necessary
+- mutates the response object
+
+As a final step, metadata about objective progress is written to the database. The transaction is then commited and the row locks are released.
+
+> Q: Which rows should be locked? One answer is: the objective row and all of the channel rows in the scope of the objective.
+
+An objective will often be blocked on, or waiting for, external events. The wallet will track the length of time an objective is blocked, and expose this information to the app. This allows the app to choose to spawn a new objective to displace the stalled one. For example, a `SubmitChallenege` objective might displace a `CloseChannel` objective.
+
+> Q: does the app have to first reject the stalled objective? It's scope could be different to the new objective's, so it seems like the only safe thing to do is to release the _entire_ scope of the stalled objective.
+
+### Termination
+
+Objectives can terminate during the run loop (see below), and The App is informed via an event or the return value of an API call triggering that loop.
+
+# The Run loop
+
+The _run loop_ of the wallet looks like this:
+
+1.  Handle an API call, or hear a blockchain event: get all state updates in the store.
+2.  UpdateFunding (for any channels that have changed)
+3.  CrankToCompletion (for any objectives whose scoped channels have been ‘touched’).
+4.  Repeat 2-3 until no more changes
+5.  ProcessLedgerQueue (see below)
+6.  Repeat 2-5 until no more changes.
+7.  Return consolidated updates and states if an API call.
+
+# Updating Funding
+
+TODO
+
```

Comment:
As I said in the PR description, it is my preference that this document is maintained as a record of what we want the wallet to be, rather than what it is.